app/views/products/_form.html.erb

```diff
@@ -1,18 +1,12 @@
 <div class="w-50 d-flex justify-content-between align-items-center pb-5">
-<%= form_with(model: [@variant, product], local: true) do |f| %>
-  <%= f.label :title %>
-  <%= f.text_area :title %>
-    <%= f.label :description %>
-
-    <%= f.text_area :description %>
-
-    <h1> Variants</h1>
-    <%= f.label :sku %>
-    <%= f.text_area :sku %>
-    <%= f.label :stock_quantity %>
-    <%= f.number_field :stock_quantity %>
-    <%= link_to 'Add Variant', new_product_variant_path(product), class: "btn btn-primary ml-5"  %>
-
+<%= simple_form_for(@product) do |f| %>
+  <%= f.input :title %>
+  <%= f.input :description %>
+  <%= link_to 'Add Variant', new_product_variant_path(@product), class: "btn btn-primary ml-5"  %>
```

Comment:
At the moment you cannot load the page because you get an error on this line. You can't have a nested URL on a resource which is not persisted in the database yet because the URL would require its ID, e.g. `/products/:id/variants/new`.

This button should add new variant row in the form using JavaScript thus you won't need this URL at all. Persisting the data should happen in a single endpoint by specification. This would make the whole `VariantsController` unnecessary.